libs/architectures/mlpe/architectures/flows/flow.py

```diff
@@ -73,12 +73,14 @@ def to_device(self, device):
 
     @property
     def context_dim(self):
-        sample_tensor = torch.zeros((1, self.n_ifos, self.strain_dim))
-        context_dim = self.embedding_net(sample_tensor).shape[-1]
+        dummy_tensor = torch.zeros((1, self.n_ifos, self.strain_dim))
+        context_dim = self.embedding_net(dummy_tensor).shape[-1]
         return context_dim
 
     @property
     def flow(self):
         if self._flow is None:
-            self.build_flow()
```

Comment:
OK with this change. But I'm curious why this is better choice?

Reply:
This original code acted like a singleton. If the flow was built, it would give the flow object, else build the flow and returned it.

Reply:
This is a relic of when I had a setup where build_flow() took the `embedding` as an argument. 

I can change it back.